### PR TITLE
agents: fix gnd person

### PR DESCRIPTION
**Pull request by rerowep**
* Only uses records with 075 $b p/b/f for creating of GND records.
* Closes https://github.com/rero/rero-ils/issues/2803.

### CODE REVIEW

**Review by PascalRepond — APPROVED** [review]

File: rero_mef/marctojson/do_gnd_agent.py [hunk 1]
```diff
@@ -45,17 +54,41 @@ def __init__(self, marc, logger=None, verbose=False, transform=True):
...
+        - f Konferenz
+        - g Geografikum
+        - n Person (nicht individualisiert)
+        - p Person (individualisiert)
+        - s Sachbegriff
+        - u Werk
+        """
+        for field_075 in self.marc.get_fields('075') or []:
+            if field_075['2'] == 'gndgen':
+                return RECORD_TYPES.get(field_075['b'])
```
Comment (zannkukai) [inline rero_mef/marctojson/do_gnd_agent.py:71]: What's happens with MARCXML python library if subfield '2' or 'b' isn't present ? Is a `KeyError` exception is raised ? If yes, this exception should be catched to return default/None value

File: rero_mef/marctojson/do_gnd_agent.py [hunk 2]
```diff
@@ -45,17 +54,41 @@ def __init__(self, marc, logger=None, verbose=False, transform=True):
...
-                self.marc.get_fields('111'):
-            for func in dir(self):
-                if func.startswith('trans'):
-                    func = getattr(self, func)
-                    func()
+        record_type = self.get_type()
+        if record_type in ['bf:Person', 'bf:Organisation']:
+            if self.marc.get_fields('100') or \
+                    self.marc.get_fields('110') or \
+                    self.marc.get_fields('111'):
```
Comment (zannkukai) [inline rero_mef/marctojson/do_gnd_agent.py:79]: from doc (https://pymarc.readthedocs.io/en/latest/#pymarc.record.Record.get_fields)
It should work,  isn't it ?
```python
if self.marc.get_fields('100', '110', '111'):
    ...
```

File: rero_mef/marctojson/do_gnd_agent.py [hunk 3]
```diff
@@ -45,17 +54,41 @@ def __init__(self, marc, logger=None, verbose=False, transform=True):
...
+                for func in dir(self):
+                    if func.startswith('trans'):
+                        func = getattr(self, func)
+                        func()
```
Comment (zannkukai) [inline rero_mef/marctojson/do_gnd_agent.py:83]: could be `getattr(self, func)()` but I agree it's not perfectly readable

File: rero_mef/marctojson/do_gnd_agent.py [hunk 4]
```diff
@@ -31,6 +31,15 @@
     'u': 'Unknown'
 }
 
+RECORD_TYPES = {
+    'p': 'bf:Person',
+    'b': 'bf:Organisation',
```
Comment (jma) [inline rero_mef/marctojson/do_gnd_agent.py:36]: Is it normal that the both lines are identical?

**Review by jma — APPROVED** [review]

File: rero_mef/marctojson/do_gnd_agent.py (hunk 4 above)
Comment (rerowep) [reply rero_mef/marctojson/do_gnd_agent.py:36]: 'b' and 'f' are organisations.

File: rero_mef/marctojson/do_gnd_agent.py (hunk 1 above)
Comment (rerowep) [reply rero_mef/marctojson/do_gnd_agent.py:71]: No exception here:
https://github.com/edsu/pymarc/blob/master/pymarc/field.py

**Pull request merged**